Client/Frontend/Browser/PageZoom/PageZoomSettingsView.swift

```diff
@@ -0,0 +1,170 @@
+          
+          let zoomLevel = Int(step * 100.0)
+          let preferredZoomLevel = Int(defaultZoomLevel * 100.0)
+         
+          Button(action: {
+            Preferences.General.defaultPageZoomLevel.value = step
+            defaultZoomLevel = step
+            
+            domains = PageZoomSettingsView.filteredDomains(context)
+          }, label: {
+            HStack(spacing: 0.0) {
+              Text(PageZoomView.percentFormatter.string(from: .init(value: step)) ?? "\(zoomLevel)%")
+                .foregroundColor(Color(UIColor.braveLabel))
+                .multilineTextAlignment(.trailing)
+              
+              Spacer()
+              
+              if preferredZoomLevel == zoomLevel {
+                Image(systemName: "checkmark")
+                  .foregroundColor(Color(UIColor.braveLabel))
+              }
+            }
+          })
+        }
+      }
+      .listRowBackground(Color(UIColor.secondaryBraveGroupedBackground))
+      
+      if !domains.isEmpty {
+        Section(
+          header: Text(Strings.PageZoom.resetAllDescription)
+            .font(.footnote)
+            .foregroundColor(Color(UIColor.secondaryBraveLabel))
+        ) {
+          Button(Strings.PageZoom.resetAll) {
+            PageZoomSettingsView.allDomains(context).forEach({
+              $0.zoom_level = nil
+            })
+            
+            try? context.save()
+            
+            domains = PageZoomSettingsView.filteredDomains(context)
+            
+            Preferences.General.defaultPageZoomLevel.notify()
+          }
+        }
+        .listRowBackground(Color(UIColor.secondaryBraveGroupedBackground))
+      }
+    }
+    .navigationTitle(Strings.PageZoom.settingsTitle)
+    .navigationBarTitleDisplayMode(.inline)
+    .listStyle(.insetGrouped)
+  }
+  
+  private func resetItemZoomLevel(at offsets: IndexSet) {
+    offsets.forEach({
+      domains[$0].zoom_level = nil
+    })
+    
+    try? context.save()
+    
+    domains = PageZoomSettingsView.filteredDomains(context)
+    
+    Preferences.General.defaultPageZoomLevel.notify()
+  }
+  
+  private static func allDomains(_ context: NSManagedObjectContext) -> [Domain] {
+    let request = NSFetchRequest<Domain>().then {
+      $0.entity = Domain.entity(context)
+      $0.sortDescriptors = [NSSortDescriptor(key: "url", ascending: true)]
+    }
+    
+    do {
+      let domains = try context.fetch(request)
+      return domains.sorted(by: { $0.host < $1.host })
+    } catch {
+      return []
+    }
+  }
+  
+  private static func filteredDomains(_ context: NSManagedObjectContext) -> [Domain] {
+    let request = NSFetchRequest<Domain>().then {
+      $0.entity = Domain.entity(context)
+      $0.sortDescriptors = [NSSortDescriptor(key: "url", ascending: true)]
+      $0.predicate = NSPredicate(format: "zoom_level != nil AND zoom_level != %@", NSNumber(value: Preferences.General.defaultPageZoomLevel.value))
```

Comment:
Wouldn't you want to show overrides for all websites regardless of if it matches the default zoom level?

Reply:
No because if the default zoom level is 125%, it will show `125%` for all websites if they have that zoom level. The idea here is that if the two levels match, just don't show it.

So instead of having a bunch of websites with `125%` showing, just only show the ones that aren't the same as the default.